src/compiler/checker.ts

```diff
@@ -15484,7 +15489,11 @@ namespace ts {
                     }
                 }
 
-                reportError(message, sourceType, targetType);
+                if (isLiteralType(source) && !(source.flags & TypeFlags.Union) && !isLiteralType(target) && !(target.flags & TypeFlags.Union)) {

```

Comment:
The problem before was that you were doing the work when there were some unit types, which meant that you might end up with some compatible base types in the target.

How about a heuristic where you say "only use the base types if the target has no unit types at all"?

```suggestion
                if (isLiteralType(source) && !forEachType(target, isUnitType))
```

Reply:
SGTM, I like that a lot. Thanks!